compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -238,13 +241,24 @@ impl<'tcx> ConstToPat<'tcx> {
                     _ => {}
                 }
             }
+
+            // Always check for `PartialEq`, even if we emitted other lints. (But not if there were
+            // any errors.) This ensures it shows up in cargo's future-compat reports as well.
+            if !self.type_has_partial_eq_impl(cv.ty()) {
```

Comment:
There is the potential alternative of checking for `Eq` here instead of `PartialEq`. This would then subsume the "disallow floats in patterns" lint as well. However with t-lang in the past having been [disinclined to disallow float patterns](https://github.com/rust-lang/rust/pull/84045), it's not clear that we *want* to require `Eq`.